src/test/TestHelper.cpp

```diff
@@ -25,13 +25,13 @@ bool adaptive::AdaptiveTree::download(const char* url,
                                       void* opaque,
                                       bool scanEffectiveURL)
 {
-  FILE* f = fopen(testFile.c_str(), "rb");
+  FILE* f = fopen(testHelper::testFile.c_str(), "rb");
```

Comment:
Maybe not for now either but Kodi provides a file class for addons that can be used to read files.

E.g. from iptvsimple using this: https://github.com/kodi-pvr/pvr.iptvsimple/blob/292ce5f4bf3a59917677de0473cb175ac3e0b84a/src/iptvsimple/utilities/FileUtils.cpp#L49-L61

Reply:
The big problem with wanting to do this is that we need an instance of Kodi to use it. It'd be wonderful to have a mock instance to use but I think building something like that is out of my depth.

Reply:
Of course. That makes perfect sense!